src/tools/remote-test-client/src/main.rs

```
@@ -116,11 +165,14 @@ fn push(path: &Path) {
     let client = t!(TcpStream::connect("127.0.0.1:12345"));
     let mut client = BufWriter::new(client);
     t!(client.write_all(b"push"));
-    t!(client.write_all(path.file_name().unwrap().to_str().unwrap().as_bytes()));
-    t!(client.write_all(&[0]));
-    let mut file = t!(File::open(path));
-    t!(io::copy(&mut file, &mut client));
+    send(path, &mut client);
     t!(client.flush());
+
+    // Wait for an acknowledgement that all the data was received. No idea
+    // why this is necessary, seems like it shouldn't be!
```

Ah yeah I ended up describing this in a comment I just wrote, it's related to the "ignore errors on accept sorta" path as well. 

Although I guess this is a facet of TCP and not of the emulated network! I'll check here that `ack ` was sent though to be extra sure.